tests/e2e/tkgs_ha.go

```diff
@@ -488,4 +344,785 @@ var _ = ginkgo.Describe("[csi-tkgs-ha] Tkgs-HA-SanityTests",
+				pvcSpec, metav1.CreateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Wait for SV PVC to come to bound state")
+			svcPv, err := fpv.WaitForPVClaimBoundPhase(svClient, []*v1.PersistentVolumeClaim{svPvclaim},
+				framework.ClaimProvisionTimeout)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			volumeID := svPvclaim.Name
+			staticPVLabels := make(map[string]string)
+			staticPVLabels["fcd-id"] = volumeID
+
+			// Get allowed topologies for zonal storage
+			allowedTopologies := getTopologySelector(allowedTopologyHAMap, categories,
+				tkgshaTopologyLevels)
+
+			ginkgo.By("Creating the PV")
+			staticPv := getPersistentVolumeSpecWithStorageClassFCDNodeSelector(volumeID,
+				v1.PersistentVolumeReclaimRetain, storageclass.Name, staticPVLabels,
+				diskSize, allowedTopologies)
+			staticPv, err = client.CoreV1().PersistentVolumes().Create(ctx, staticPv, metav1.CreateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Creating the PVC")
+			staticPvc := getPersistentVolumeClaimSpec(namespace, staticPVLabels, staticPv.Name)
+			staticPvc.Spec.StorageClassName = &storageclass.Name
+			staticPvc, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(ctx, staticPvc, metav1.CreateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			// Wait for PV and PVC to Bind.
+			framework.ExpectNoError(fpv.WaitOnPVandPVC(client, framework.NewTimeoutContextWithDefaults(),
+				namespace, staticPv, staticPvc))
+
+			defer func() {
+				err := fpv.DeletePersistentVolumeClaim(client, staticPvc.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				err = fpv.DeletePersistentVolume(client, staticPv.Name)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				ginkgo.By("Verify PVs, volumes are deleted from CNS")
+				err = fpv.WaitForPersistentVolumeDeleted(client, staticPv.Name, framework.Poll,
+					framework.PodDeleteTimeout)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeID)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+					fmt.Sprintf("Volume: %s should not be present in the CNS after it is deleted from "+
+						"kubernetes", volumeID))
+				ginkgo.By("Verify volume is deleted in Supervisor Cluster")
+				volumeExists := verifyVolumeExistInSupervisorCluster(svcPv[0].Spec.CSI.VolumeHandle)
+				gomega.Expect(volumeExists).To(gomega.BeFalse())
+
+			}()
+
+			ginkgo.By("Verify SV storageclass points to GC storageclass")
+			gomega.Expect(*svPvclaim.Spec.StorageClassName == storageclass.Name).To(
+				gomega.BeTrue(), "SV storageclass does not match with gc storageclass")
+			framework.Logf("GC PVC's storageclass matches SVC PVC's storageclass")
+
+			ginkgo.By("Verify GV PV has has required PV node affinity details")
+			_, err = verifyVolumeTopologyForLevel5(staticPv, allowedTopologyHAMap)
```

Comment:
"verifyVolumeTopologyForLevel5" - You have made this as a generic code for both TKGS-HA and topology level 5 tests. Since the code is generic now , this name does not suite. Please note this and we need to change this method name . 
Not in this PR , because it require to change in many places. Please note this and make the change in a separate PR